src/main/haskell/kore/src/Kore/Step/Step.hs

```diff
@@ -178,3 +305,71 @@ heatingCooling axiomStrategy axioms =
     normal = axiomStrategy normalRules
     coolingRules = filter isCoolingRule axioms
     cool = axiomStrategy coolingRules
+
+{-| A strategy for doing the first step of a one-path verification.
+For subsequent steps, use 'onePathFollowupStep'.
+
+It first removes the destination from the input, then it tries to apply
+the normal axioms.
+
+Whenever it applies an axiom, the subsequent axioms see only the part of the
+pattern to which the axiom wasn't applied.
+-}
+onePathFirstStep
+    :: axiom
+    -- ^ axiom with which to remove the destination we`re trying to reach
+    -> [axiom]
+    -- ^ normal axioms
+    -> Strategy (Prim axiom)
+onePathFirstStep destinationRemovalAxiom axioms =
+    Strategy.sequence
+        (  [axiomWithRemainderStep destinationRemovalAxiom]
+        ++ map
+            axiomOnRemainderStep
+            axioms
+        ++ [Strategy.apply nonEmptyRemainderFail]
```

Comment:
Why should the step fail on a non-empty remainder? I guess this question might answer itself if I understood what the remainder was about.

Reply:
I added a comment, but https://github.com/kframework/kore/blob/master/docs/2018-11-08-One-Path-Reachability-Proofs.md may help more, especially the "applying the axiom" section.

Reply:
I was and am reading that document, but some connections are still unclear. For example, the terminology is different; I'm still not sure what corresponds to the "remainder" in that document. Another thing that is not clear is that everything in that document is conjoined at the top level, but in the code it is disjoined.

Reply:
Added explanations for both.